packages/amplify-category-api/src/provider-utils/awscloudformation/utils/rds-resources/database-resources.ts

```diff
@@ -151,3 +159,34 @@ export const getDatabaseName = async (context: $TSContext, apiName: string, secr
 
   return secrets[0].secretValue;
 };
+
+export const removeVpcSchemaInspectorLambda = async (context) => {
+  
+  try {
+    // Delete the lambda function
+    const meta = stateManager.getMeta();
+    const { AmplifyAppId, Region } = meta.providers.awscloudformation;
+    const { amplify } = context;
+    const { envName } = amplify.getEnvInfo();
+    const lambdaName = getVpcMetadataLambdaName(AmplifyAppId, envName);
+
+    const client = new LambdaClient({ region: Region });
+    const command = new DeleteFunctionCommand({ FunctionName: lambdaName });
+    await client.send(command);
+
+    // Delete the role and policy
+    await deleteSchemaInspectorLambdaRole(lambdaName);
+  }
+  catch (error) {
+    printer.debug(`Error deleting the schema inspector lambda: ${error}`);
+    // 1. Ignore if the AppId is not found error.
+    // 2. Schema introspection will exist only on databases imported from VPC. Ignore the error on environment deletion.
```

Comment:
Is this WIP code, or a suggestion for the user?

Reply:
This is the reasoning for not throwing an exception while cleaning up resources.